spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.binding;
+
+import org.springframework.messaging.MessageChannel;
+
+/**
+ * Defines methods to create/configure the {@link org.springframework.messaging.MessageChannel} defined
+ * in {@link org.springframework.cloud.stream.annotation.EnableBinding}.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public interface ChannelFactory {
+
+	String SPRING_CLOUD_STREAM_INTERNAL_PREFIX = "spring.cloud.stream.internal";
```

Comment:
Let's place the property names elsewhere (maybe in the implementation). `ChannelFactory` should only define the contract in `createChannel`.